docs/src/main/java/sample/jpa/service/authorization/JpaOAuth2AuthorizationService.java

```diff
@@ -90,29 +90,64 @@ public OAuth2Authorization findByToken(String token, OAuth2TokenType tokenType)
-		} else if (OAuth2ParameterNames.STATE.equals(tokenType.getValue())) {
-			result = this.authorizationRepository.findByState(token);
-		} else if (OAuth2ParameterNames.CODE.equals(tokenType.getValue())) {
-			result = this.authorizationRepository.findByAuthorizationCodeValue(token);
-		} else if (OAuth2ParameterNames.ACCESS_TOKEN.equals(tokenType.getValue())) {
-			result = this.authorizationRepository.findByAccessTokenValue(token);
-		} else if (OAuth2ParameterNames.REFRESH_TOKEN.equals(tokenType.getValue())) {
-			result = this.authorizationRepository.findByRefreshTokenValue(token);
-		} else if (OidcParameterNames.ID_TOKEN.equals(tokenType.getValue())) {
-			result = this.authorizationRepository.findByOidcIdTokenValue(token);
-		} else if (OAuth2ParameterNames.USER_CODE.equals(tokenType.getValue())) {
-			result = this.authorizationRepository.findByUserCodeValue(token);
-		} else if (OAuth2ParameterNames.DEVICE_CODE.equals(tokenType.getValue())) {
-			result = this.authorizationRepository.findByDeviceCodeValue(token);
-		} else {
-			result = Optional.empty();
+
+		try {
+			if (tokenType == null) {
+				result = this.authorizationRepository.findByStateOrAuthorizationCodeValueOrAccessTokenValueOrRefreshTokenValueOrOidcIdTokenValueOrUserCodeValueOrDeviceCodeValue(token);
+			} else if (OAuth2ParameterNames.STATE.equals(tokenType.getValue())) {
+				result = this.authorizationRepository.findByState(token);
+			} else if (OAuth2ParameterNames.CODE.equals(tokenType.getValue())) {
+				result = this.authorizationRepository.findByAuthorizationCodeValue(token);
+			} else if (OAuth2ParameterNames.ACCESS_TOKEN.equals(tokenType.getValue())) {
+				result = this.authorizationRepository.findByAccessTokenValue(token);
+			} else if (OAuth2ParameterNames.REFRESH_TOKEN.equals(tokenType.getValue())) {
+				result = this.authorizationRepository.findByRefreshTokenValue(token);
+			} else if (OidcParameterNames.ID_TOKEN.equals(tokenType.getValue())) {
+				result = this.authorizationRepository.findByOidcIdTokenValue(token);
+			} else if (OAuth2ParameterNames.USER_CODE.equals(tokenType.getValue())) {
+				result = this.authorizationRepository.findByUserCodeValue(token);
+			} else if (OAuth2ParameterNames.DEVICE_CODE.equals(tokenType.getValue())) {
+				result = this.authorizationRepository.findByDeviceCodeValue(token);
+			} else {
+				result = Optional.empty();
+			}
+		}catch (Exception e){
+			handleException(token, tokenType, e);
+			return null;
 		}
 
 		return result.map(this::toObject).orElse(null);
 	}
 
+	private void handleException(String token, OAuth2TokenType tokenType, Exception e) {
+
+		if (this.logger.isWarnEnabled()) {
+			this.logger.warn(LogMessage.format("Failed in running findByToken (token: '%s', error details: '%s')", token, e.getMessage()));
+		}
+
+		String value = tokenType != null ? tokenType.getValue() : null;
+
+		if (OAuth2ParameterNames.STATE.equals(value)) {
+			this.authorizationRepository.deleteAllByState(token);
```

Comment:
@Andrew-Kang-G 

We cannot assume the error condition is directly related to "duplicate results":

> findByToken has the potential to cause errors, such as returning duplicate results

Also keep in mind that this is sample code that we have provided in a How-to guide, and the expectation is that the user will take the code and enhance it to fit the requirements for their application.

I'm going to close this as I don't think we should introduce the `delete**` operations as we want to keep the sample a simple starting point for users.

Reply:
Thank you for the answer. @jgrandja 

For normal operations, duplicate results don't occur. However, when considering a scenario where a user is using multiple devices, their access tokens should be managed separately. In such cases, the access token values could potentially be the same, as they are encoded blobs consisting of the user's information. This means there is a possibility of collisions. If the library's standard implementation ensures that the same token is never generated at any time, I agree. 

| principal_name | registered_client_id | access_token_value |customized_my_value sent from each mobile device|
|-------|-------|-------|-------|
| customer1 |app_customer| SameBlobRRR|mobile-Android-Chris...|
| customer1 |app_customer| SameBlobRRR|mobile-Android-Jane...|
| customer2 |app_customer| DifferentBlobRRR|mobile-Android-DDD |